src/components/NanoContract/NanoContractTransactionActionListItem.js

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import {
+  TouchableHighlight,
+  StyleSheet,
+  View,
+  Text,
+} from 'react-native';
+import { useSelector } from 'react-redux';
+import { t } from 'ttag';
+import { NANO_CONTRACT_ACTION } from '../../constants';
+import { COLORS } from '../../styles/themes';
+import { getShortHash, isTokenNFT, renderValue } from '../../utils';
+import { ReceivedIcon } from '../Icons/Received.icon';
+import { SentIcon } from '../Icons/Sent.icon';
+
+/**
+ * Retrives token symbol, otherwise returns a shortened token hash.
+ *
+ * @param {string} tokenUid Token hash
+ * @returns {(state) => boolean} Callback that takes state as input
+ *
+ * Remarks:
+ * This function should be used combined with `useSelector`.
+ */
+function getTokenSymbol(tokenUid) {
+  return (state) => {
+    const tokens = state.tokens || {};
+    if (tokenUid in tokens) {
+      return tokens[tokenUid].symbol;
+    }
+    return getShortHash(tokenUid, 7);
+  };
+}
+
+/**
+ * Checks if the referred token is an NFT.
+ *
+ * @param {string} tokenUid Token hash
+ * @returns {(state) => boolean} Callback that takes state as input
+ *
+ * Remarks:
+ * This function should be used combined with `useSelector`.
+ */
+function checkIsTokenNft(tokenUid) {
+  return (state) => isTokenNFT(tokenUid, state.tokenMetadata || {});
+}
+
+/**
+ * It renders the item of actions list of a Nano Contract transaction.
+ *
+ * @param {Object} props
+ * @param {{
+ *   type: string;
+ *   uid: string;
+ *   amount: number;
+ * }} props.item A transaction action
+ */
+export const NanoContractTransactionActionListItem = ({ item }) => {
+  const tokenSymbol = useSelector(getTokenSymbol(item.uid));
+  const isNft = useSelector(checkIsTokenNft(item.uid));
+
+  return (
+    <Wrapper>
+      <Icon type={item.type} />
+      <ContentWrapper tokenSymbol={tokenSymbol} type={item.type} />
+      <TokenAmount amount={item.amount} isNft={isNft} type={item.type} />
+    </Wrapper>
+  );
+};
+
+const Wrapper = ({ children }) => (
+  <TouchableHighlight underlayColor={COLORS.primaryOpacity30}>
```

Comment:
Why a `TouchableHighlight` here?

Reply:
It was used by default. I think it doesn't hurt, but do you think it should be removed?